homeassistant/components/mqtt/__init__.py

```diff
@@ -198,6 +215,20 @@ def valid_publish_topic(value: Any) -> str:
                  default=DEFAULT_PAYLOAD_NOT_AVAILABLE): cv.string,
 })
 
+MQTT_ENTITY_DEVICE_INFO_SCHEMA = vol.All(vol.Schema({
+    vol.Optional(CONF_IDENTIFIERS, default=list):
+        vol.All(cv.ensure_list, [cv.string]),
+    vol.Optional(CONF_CONNECTIONS, default=list):
+        vol.All(cv.ensure_list, [vol.Schema({
+            vol.Required(CONF_TYPE): vol.Any(*device_registry.CONNECTION_KEYS),
+            vol.Required(CONF_IDENTIFIER): cv.string,
+        })]),
+    vol.Optional(CONF_MANUFACTURER): cv.string,
+    vol.Optional(CONF_MODEL): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_SW_VERSION): cv.string,
+}), validate_device_has_at_least_one_identifier)
```

Comment:
As the `device` parameter only really does something if `unique_id` is also set AND the config is sent via MQTT discovery, should I implement a custom validator checking that both conditions hold true? Or is that not necessary.

Reply:
Just mention it in the docs? Or maybe we should print a warning? Not sure.

Reply:
There is `cv.has_at_least_one_key`.

Reply:
Added it to the docs. @MartinHjelmare well the condition is "`device` in config => `unique_id` in config", unique_id without device is ok and neither unique_id nor device is also ok.

To clarify, writing the validator is not the problem (I've learned enough about that at https://github.com/OttoWinter/esphomeyaml 😉), but it would just end up making the `PLATFORM_SCHEMA` in all platforms with this feature a bit less nice (more indentation, extra line for `vol.All`). I guess the docs are good enough.

Reply:
Are you saying you will remove this validator?

If we still need to validate at least one identifier, we can use `cv.has_at_least_one_key` to do that.

Reply:
Ah sorry, I thought you were referring to the `unique_id`/`device` thing. 

The `validate_device_has_at_least_one_identifier` validator checks if at least one "identifier" exists in the device config. These identifiers can either be pure `identifiers` or `connections`. 

`has_at_least_one_key` checks if at least one of the specified keys exist. The custom validator `validate_device_has_at_least_one_identifier` checks if the `identifiers` and `connections` lists at least contain one **item**.

I plan on keeping this validator. The validator I will not add is the one checking if `device` exists together with `unique_id`.

Reply:
You're correct!